### ISSUE TITLE
NewDialer() should support a hostname as an argument for SMTP helo

### ISSUE THREAD

**Issue by phocean** [go-gomail/gomail#62]
Hi,

As you state in the comments, e-mails are sent with "localhost" as the default hostname during the SMTP helo stage.

It causes some trouble as many anti-spam solution would badly penalize any e-mail sent by a machine presenting itself as "localhost".

Please update the function so that it takes the hostname as an argument.

```
func NewDialer(host string, port int, username, password string) *Dialer {
        return &Dialer{
                Host:     host,
                Port:     port,
                Username: username,
                Password: password,
                SSL:      port == 465,
                LocalName: "test.domain",
        }
}
```

**Comment by jordan-wright** [comment]
It would probably work to just use the configured hostname, right? I'm assuming on production systems the hostname is set appropriately.

**Comment by phocean** [comment]
Sure, but I am on a production system with a correctly configured hostname.

Moreover, I checked the code and, although I had never looked at go code before, I do not see it getting the hostname anywhere. I could be wrong, of course.

**Comment by jordan-wright** [comment]
Yes, I know, I was referring to using the hostname in the repo instead of "localhost".
